### PR TITLE
Expose data files rows statistics fields in $manifests table

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Expose `org.apache.iceberg.ManifestFile` data files rows fields  which are not yet exposed in the `$manifests` metadata table.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Iceberg connector

> How would you describe this change to a non-technical end user or system administrator?

Expose further fields in the `$manifests` Iceberg metadata table.

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Iceberg
* Add `added_rows_count`, `existing_rows_count` and `deleted_rows_count` columns to `$manifests` table
```

### CODE REVIEW

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java [hunk 1]
```diff
@@ -64,6 +64,19 @@ public void setUp()
...
+                "INSERT INTO test_schema.test_table_with_dml " +
+                        "VALUES " +
+                        "('a1', DATE '2022-01-01'), ('a2', DATE '2022-01-01'), " +
+                        "('b1', DATE '2022-02-02'), ('b2', DATE '2022-02-02'), " +
+                        "('c1', DATE '2022-03-03'), ('c2', DATE '2022-03-03')",
+                6);
+        assertUpdate("UPDATE test_schema.test_table_with_dml SET _varchar = 'a1.updated' WHERE _date = DATE '2022-01-01' AND _varchar = 'a1'", 1);
+        assertUpdate("DELETE FROM test_schema.test_table_with_dml WHERE _date = DATE '2022-02-02' AND _varchar = 'b2'", 1);
+        assertUpdate("INSERT INTO test_schema.test_table_with_dml VALUES ('c3', DATE '2022-03-03'), ('d1', DATE '2022-04-04')", 2);
+        assertQuery("SELECT count(*) FROM test_schema.test_table_with_dml", "VALUES 7");
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java:79]: What's the relationship between new columns in `$manifests` and this test change?

**Review by ebyhr — COMMENTED** [review]
Could you please update `iceberg.rst`?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java (hunk 1 above)
Comment (findinpath) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java:79]: This change showcases in further detail how the data files row statistic fields are filled in a DML(UPDATE/DELETE) scenario.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**